docs/classes.rst

```diff
@@ -210,6 +211,59 @@ There are already some utility classes implemented in ``pytest-mh``. See
     Each change that is made through the utility object (such as writing to a
     file) is automatically reverted (the original file is restored).
 
+TopologyController
+==================
+
+Topology controller can be assigned to a topology via `@pytest.mark.topology`
+or through known topology class. This controller provides various methods to
+control the topology behavior:
+
+* per-topology setup and teardown, called once before the first test/after the
+  last test for given topology is executed
+* per-test topology setup and teardown, called before and after every test case
+  for given topology
+* check topology requirements and skip the test if these are not satisfied
+
+In order to use the controller, you need to inherit from
+:class:`~pytest_mh.TopologyController` and override desired methods. Each method
+can take any parameter as defined by the topology fixtures. The parameter value
+is an instance of a :class:`~pytest_mh.MultihostHost` object.
+
+See :class:`~pytest_mh.TopologyController` for API documentation
+
+.. code-block: python
+    :caption: Topology controller example
```

Comment:
Is it correct that each of these lines below go into the test .py file itself?

Reply:
It's just a brief example, I don't think it is needed to split it into multiple files for an example. It is functional of course, if put into single file.

Reply:
I added this example to `sssd_test_framework/topology.py` and pushed these changes to my branch `topology_controller_example`, set this in `requirements.txt`

~~~
pytest
git+https://github.com/next-actions/pytest-importance
git+https://github.com/pbrezina/pytest-mh@topology-setup
git+https://github.com/next-actions/pytest-ticket
git+https://github.com/next-actions/pytest-tier
git+https://github.com/next-actions/pytest-output
git+https://github.com/justin-stephenson/sssd-test-framework@topology_controller_example
~~~

Then created a sssd system test with

~~~
@pytest.mark.topology(KnownTopology.EXAMPLE)
def test_topology_controller(client: Client):
~~~

but it fails here https://privatebin.corp.redhat.com/?8d25c7c1aec5c226#HGTEUCdHvqULYupMPnDKBSSsGwDXfY3qH232G18rA5Cy

Reply:
This should be fixed by https://github.com/next-actions/pytest-mh/pull/38/commits/a089db0063a2079d8dfb023e849281f96b1df5e7